### ISSUE TITLE
Action hook before redirection plugin

### ISSUE THREAD

**Issue by pasaro** [johngodley/redirection#559]
To help track problems please follow this template. If you can provide screenshots then that will help.

*Please replace this template with your own details!*

### Summary

Explain what you are trying to do. If you want to suggest a new feature then explain what you want it to do

### Actual Behavior

Explain what actually happened. If suggesting a new feature then explain what currently happens.

### Steps to reproduce

Explain the steps involved to reproduce the problem

### Environment

Please list your browser, browser version, version of WordPress and Redirection plugin.

**Comment by pasaro** [comment]
I have many old blogger permalinks that still cached on google and others directories. To redirect them to the correct posts I tried to use this function in my theme function.php.

add_action('wp-head', 'wp_redirect_blogger_oldlink');
function wp_redirect_blogger_oldlink() {
    $Rurl = $_SERVER['REQUEST_URI'];
      if (stripos($Rurl,'html') !== false) {
	$res = $wpdb->get_results("SELECT post_id, meta_value FROM $wpdb->postmeta WHERE meta_key = 'blogger_permalink'");
	if ($res) {
        $Aurl= strstr($_SERVER[REQUEST_URI], '_', true);
        $Burl = strstr($_SERVER[REQUEST_URI], '.html', true);
	   foreach ($res as $row) {
		if ( stripos($row->meta_value, $Burl) !== false || stripos($row->meta_value, $Aurl) !== false ) {
			wp_redirect( get_permalink($row->post_id), 301 );
			die();
		} 
     	   } 
        }
      }
   }

The function failed to work in my theme function.php but can works well when I put it in the 404.php, which is in redirection plugin log report counts as 404 reports.

It seems like I have to use a right action hook to trigger the function "BEFORE" the redirection plugin in the theme function.php.

Any ideas how to solve this problem?

Thanks in advance.

**Comment by johngodley** [comment]
I'm not sure I understand how this relates to Redirection?

**Comment by pasaro** [comment]
My point is I still can't integrate that function to the plugin redirection rules. If I can make similar function in the redirection plugin it won't be  404 report from blogger permalink anymore.
I have tried some regex but still cannot figured out the right code.

FYI, blogger permalinks used only  first 60 characters of post title while wordpress used the whole title as permalinks. 

The redirect guess 404 permalink from wordpress core can help some links but not all. And by adding that function I can redirect all of the old links to the right permalinks.

I saw that some old permalinks got recorded in the redirection logs (which is redirected to the right wordpress permalinks) and some others go to  404 logs.

The only way I can redirected the remaining uncaught links is by adding the function in the 404 page template. I know this is not the right way.

It will be great if all links can be redirected before counted as 404 not found links.
